src/js/plugins/tts/utils.js

```diff
@@ -0,0 +1,21 @@
+import 'es6-promise/auto';
+
+/**
+ * Waits the provided number of ms and then resolves with a promise
+ * @param {number} ms
+ * @return {PromiseLike}
+ */
+export function sleep(ms) {
+    return new Promise(res => setTimeout(res, ms));
+}
+
+/**
+ * Checks whether the current browser is a Chrome/Chromium browser
+ * Code from https://stackoverflow.com/a/4565120/2317712
+ * @param {string} [userAgent]
+ * @param {string} [vendor]
+ * @return {boolean}
+ */
+export function isChrome(userAgent=navigator.userAgent, vendor=navigator.vendor) {
```

Comment:
Is this tts specific? Or is there a more central location where shared utilities should go? @jbuckner @iisa @cdrini any thoughts?

Reply:
Functions like these are generally considered bad practice; instead of browser sniffing, feature detection is the recommended practice. But I couldn't think of a way to feature detect this bug :/ keeping this scoped here makes it less likely to be reused.